### ISSUE TITLE
Supabase self host storage buckets alway loading

### ISSUE THREAD

**Issue by phongjimmy** [supabase/supabase#15385]
## Describe the bug

Storage buckets loading

## Screenshots
<img width="727" alt="Screenshot 2023-06-27 at 18 29 56" src="https://github.com/supabase/supabase/assets/71168450/d15e0683-23bc-403c-bba0-c065de5a55e9">

## System information

- OS: Centos 7
- Browser chrome
- Docker: 23.0.1
- Supbase: Lastest commit

## Additional context

So when when direct to url api buckets it's work
https://xxx.com/api/storage/default/buckets

<img width="565" alt="Screenshot 2023-06-27 at 18 32 43" src="https://github.com/supabase/supabase/assets/71168450/29570c71-996a-4250-8d64-53697f6ca95b">

**Comment by AriaFantom** [comment]
i faced the same issue

**Comment by leopku** [comment]
Same issue

**Comment by bombillazo** [comment]
Happening to me as well 😕

**Comment by imlida** [comment]
Same issue

**Comment by isaacfink** [comment]
same issue here

**Comment by isaacfink** [comment]
When I try to create a bucket I get this error
`Failed to create bucket: new row violates row-level security policy`
The same error occurs when I try to upload to a bucket, it seems like an rls issue where for some reason the policy fails every time

**Comment by ArsHinMQ** [comment]
Same issue on local development

**Comment by sweatybridge** [comment]
I've narrowed down the studio version which broke storage UI to these commits https://github.com/supabase/supabase/compare/7a24ddd...18c0a32 (7a24ddd works but 18c0a32 doesn't)

@joshenlim tagging you to take a look since it might be related to the storage UI refactor.

**Comment by senguendk** [comment]
Can confirm. The Storage UI is working again after #15497.

**Comment by Borahm** [comment]
Same here and when I create a new bucket I get this:

![Screenshot 2023-07-02 at 02 32 22](https://github.com/supabase/supabase/assets/3512357/19aeec06-8137-4885-95f3-5140b7c7d246)

**Comment by isaacfink** [comment]
@senguendk the ui doesn't work for me on cli version 1.75.3 I still cannot create buckets, what did you mean by it's working?

**Comment by senguendk** [comment]
> @senguendk the ui doesn't work for me on cli version 1.75.3 I still cannot create buckets, what did you mean by it's working?

This issue is about the UI not showing buckets correctly. #15497 made that work again.

**Comment by joshenlim** [comment]
hey everyone! apologies about this - we've just pushed out a [fix](https://github.com/supabase/supabase/pull/15505) for this, and the latest build of the UI should now be fixed! but do let us know nonetheless if you're still running into this issue

**Comment by bombillazo** [comment]
Can confirm I can see the buckets now!

**Comment by isaacfink** [comment]
@joshenlim which cli version? do I need to remove the images or just upgrade the cli?

**Comment by isaacfink** [comment]
Doesn't work with version 1.75.3 and latest images

**Comment by isaacfink** [comment]
@bombillazo I ran the following
```
npx supabase@latest stop
npx supabase@latest db reset
npx supabase@latest start
```

Before that I removed all containers and images to make sure I don't have an old version, when I run `npx supabase@latest --version` I getb `1.75.3` so I am almost sure I am using the correct verison, when creating a bucket I get an rls error
![image](https://github.com/supabase/supabase/assets/46644190/974868d4-4fc7-4758-be65-e66f0477f849)
The first error (500) happens whenever I visit the buckets page and the second happens when creating a bucket, both public and private
I also tried creating some buckets in sql and it didn't show up, but the ui didn't load either it just said no buckets

**Comment by bombillazo** [comment]
that is very weird, can you check the docker logs for the supabase_storage, supabase_db and kong containers? seems like there is something weird in the server.

**Comment by isaacfink** [comment]
@bombillazo  I created a new issue with some additional context
https://github.com/supabase/supabase/issues/15534

**Comment by joshenlim** [comment]
i'll go ahead to close this issue for now then since the initial problem has been resolved - we can move to 15534 to further discuss the other issue raised 🙏

**Comment by alehano** [comment]
I have the same issue. I installed two self hosted version: one of them with minimal changes works fine, but the second where I changed Postgres port, container names not loading buckets.
Endpoint /api/storage/default/buckets return 500 error.
All other components works fine.

**Comment by wade-liwei** [comment]
{"level":50,"time":"2024-07-18T07:36:10.177Z","pid":1,"hostname":"b4a437526191","reqId":"req-3w","tenantId":"stub","project":"stub","reqId":"req-3w","error":{"raw":"{\"metadata\":{},\"code\":\"AccessDenied\",\"httpStatusCode\":403,\"userStatusCode\":400,\"originalError\":{},\"error\":\"Unauthorized\"}","name":"Error","message":"Unexpected token 's', ...\"  \"ism�� \"supabase-d\"... is not valid JSON","stack":"Error: Unexpected token 's', ...\"  \"ism�� \"supabase-d\"... is not valid JSON\n    at Object.AccessDenied (/app/dist/storage/errors.js:86:34)\n    at /app/dist/auth/jwt.js:110:48\n    at module.exports [as verify] (/app/node_modules/jsonwebtoken/verify.js:78:12)\n    at /app/dist/auth/jwt.js:108:32\n    at new Promise (<anonymous>)\n    at verifyJWT (/app/dist/auth/jwt.js:107:12)\n    at Object.<anonymous> (/app/dist/http/plugins/jwt.js:19:56)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"},"msg":"unable to get owner"}
{"level":40,"time":"2024-07-18T07:36:10.178Z","pid":1,"hostname":"b4a437526191","reqId":"req-3w","tenantId":"stub","project":"stub","reqId":"req-3w","type":"request","req":{"traceId":"req-3w","method":"GET","url":"/bucket","headers":{"host":"storage:5000","x_forwarded_proto":"http","x_forwarded_host":"kong","x_forwarded_port":"8000","x_real_ip":"172.23.0.9","x_client_info":"supabase-js-node/2.41.1","accept":"*/*","user_agent":"node"},"hostname":"storage:5000","remoteAddress":"172.23.0.11","remotePort":40558},"res":{"statusCode":400,"headers":{"content_type":"application/json; charset=utf-8","content_length":"196"}},"responseTime":0.6900420188903809,"msg":"stub | GET | 400 | 172.23.0.11 | req-3w | /bucket | node"}

@alehano 

I guess you have same log.

**Comment by EricL521** [comment]
Not sure if this was what fixed it because I've been playing around with the .env file a bit, but I also had the 500 error code, and this error in the server logs: `{"level":40,"time":"2025-05-06T04:01:09.887Z","pid":1,"hostname":"4ee9ac4d456c","region":"stub","reqId":"req-6y","tenantId":"stub","project":"stub","reqId":"req-6y","appVersion":"1.22.7","type":"request","req":{"region":"stub","traceId":"req-6y","method":"GET","url":"/bucket","headers":{"host":"storage:5000","x_forwarded_proto":"http","x_forwarded_host":"kong","x_forwarded_port":"8000","x_forwarded_prefix":"/storage/v1/","x_real_ip":"172.19.0.11","x_client_info":"supabase-js-node/2.49.3","accept":"*/*","user_agent":"node"},"hostname":"storage:5000","remoteAddress":"172.19.0.6","remotePort":54692},"res":{"statusCode":400,"headers":{"content_type":"application/json; charset=utf-8","content_length":"73"}},"responseTime":5.762531280517578,"error":{"raw":"{\"metadata\":{},\"code\":\"AccessDenied\",\"httpStatusCode\":403,\"userStatusCode\":400,\"originalError\":{\"metadata\":{},\"code\":\"AccessDenied\",\"httpStatusCode\":403,\"userStatusCode\":400,\"originalError\":{\"name\":\"JsonWebTokenError\",\"message\":\"invalid signature\"},\"error\":\"Unauthorized\"},\"error\":\"Unauthorized\"}","name":"Error","message":"invalid signature","stack":"Error: invalid signature\n    at Object.AccessDenied (/app/dist/internal/errors/codes.js:121:32)\n    at Object.<anonymous> (/app/dist/http/plugins/jwt.js:62:36)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","statusCode":403},"role":"anon","resources":[],"operation":"storage.bucket.list","msg":"stub | GET | 400 | 172.19.0.6 | req-6y | /bucket | node"}`

What I changed was I regenerated my `JWT_KEY`, `ANON_KEY`, and `SERVICE_ROLE_KEY` (using [this](https://supabase.com/docs/guides/self-hosting/docker#generate-api-keys)), but when generating the service_key, I had to manually change the JWT info preview from `"role": "anon"` to `"role": "service_role"`

Hope this helps

**Comment by EricL521** [comment]
Alright, took another look at this, I've found a solution that works consistently for me, although it's more of a workaround. Here's what I did:
- Start up docker containers
- Run `docker exec supabase-kong getent hosts storage`
- Edit `kong.yml` file, line 178 from `url: http://storage:5000/` to `url: http://(ip from above):5000/` (ex. `url: http://172.20.0.14:5000/`)
- Restart kong docker container

**Comment by lmoy28** [comment]
Just tried to fix this exact issue in my self hosted version, can't seem to find the info. IP fix did not work, neither did the JWT, anyone has other ways of solving this ?
